.gitlab-ci.yml

```diff
@@ -277,31 +277,26 @@ variables:
 .if_not_run_all_builds: &if_not_run_all_builds
   if: $CI_COMMIT_BRANCH != "main" && $DEPLOY_AGENT != "true" && $RUN_ALL_BUILDS != "true"
 
-# Rule to trigger test kitchen setup, run, and cleanup.
+# Rule to trigger test setup, run, and cleanup.
 # By default:
-# - on main and deploy pipelines, kitchen tests are run
-# - on branch pipelines, kitchen tests are not run
-# RUN_KITCHEN_TESTS can be set to true to force kitchen tests to be run on a branch pipeline.
-# RUN_KITCHEN_TESTS can be set to false to force kitchen tests to not run on main/deploy pipelines.
-.if_kitchen: &if_kitchen
-  if: ($CI_COMMIT_BRANCH == "main"  || $DEPLOY_AGENT == "true" || $RUN_KITCHEN_TESTS == "true") && $RUN_KITCHEN_TESTS != "false"
-
-# Rules to trigger default kitchen tests.
-# Some of the kitchen tests are run on all pipelines by default. They can only be disabled
-# by setting RUN_KITCHEN_TESTS to false.
-.if_default_kitchen: &if_default_kitchen
-  if: $RUN_KITCHEN_TESTS != "false"
+# - on main and deploy pipelines, installer tests are run
+# - on branch pipelines, installer tests are run on a subset of the OSes we test
+# RUN_E2E_TESTS can be set to on to force all the installer tests to be run on a branch pipeline.
```

Comment:
As `RUN_E2E_TESTS` enforces `installer_tests`, should we name it `RUN_ALL_E2E_TESTS`? I'm not sure if there are dedicated variables for installer/not installer e2e tests

Reply:
We had dedicated variables for E2E and installer tests `RUN_E2E_TESTS` and `RUN_KITCHEN_TESTS` the goal of this PR is to merge the two. I think we no longer want to make a difference between what we call E2E tests and the installer tests that are also E2E tests

Reply:
I think a good compromise is to keep the variable `RUN_E2E_TESTS` but rename the alias. `if_run_e2e_tests` into `if_run_all_e2e_tests`. Because the consequence of having `RUN_E2E_TESTS` to `on` is that we run all the e2e tests